flask_cors/__init__.py

```diff
@@ -298,6 +312,19 @@ def _reg_length(maybe_regex):
     else:
         raise ValueError("Unexpected value for resources argument.")
 
+def _stringify_regexp(regexp):
+    '''
+        Helper that returns regexp pattern from given value.
+
+        :param regexp: regular expression to stringify
+        :type regexp: _sre.SRE_Pattern or str
+        :returns: string representation of given regexp pattern
+        :rtype: str
+    '''
+    try:
+        return regexp.pattern
+    except AttributeError:
+        return str(regexp)

```

Comment:
I love the idea of printing out user-compiled regexs as the original pattern for easier reading. Two quick ideas I would love your feedback on:

1.  I did a similar check using the RegexObject type which I sniff at runtime (since the type is not actually part of the Public API). I don't feel strongly as to whether we use the 'ask for forgiveness' pattern of catching the exception, or checking the instance type, but I think we should be consistent

    ```Python
         def _reg_length(maybe_regex):
             if isinstance(maybe_regex, RegexObject):
                 return len(maybe_regex.pattern)
             else:
                 return len(maybe_regex)
    ```

2. If we change this function to be called '_get_regexp_pattern', It can be shared in both places, which is wonderful. If you would be willing to name-change, and update the usage in parsing resources to be:

```Python
         # Sort by the String length, or regex pattern length if compiled
         pattern_length = lambda pair: len(_get_regexp_pattern(pair[0]))
         return sorted(resources,
                       key=pattern_length,
                       reverse=True)
```

Since this one is a bit more involved, I am happy to take on the work of refactoring these two, since I don't want it to hold you up!